luaui/Widgets/unit_smart_select.lua

```diff
@@ -89,6 +96,10 @@ for udid, udef in pairs(UnitDefs) do
 	local builder = (udef.canReclaim and udef.reclaimSpeed > 0)  or  (udef.canResurrect and udef.resurrectSpeed > 0)  or  (udef.canRepair and udef.repairSpeed > 0) or (udef.buildOptions and udef.buildOptions[1])
 	local building = (isMobile == false)
 	local combat = (not builder) and isMobile and (#udef.weapons > 0)
+	local isUtil = udef.customParams and udef.customParams.unitgroup == "util"
+	local antinuke = udef.customParams and udef.customParams.unitgroup == "antinuke"
```

Comment:
`customParams` can never be nil.
```suggestion
	local isUtil = udef.customParams.unitgroup == "util"
	local antinuke = udef.customParams.unitgroup == "antinuke"
```

Reply:
Engine always provides the table, even if the input def doesn't.